src/widgets/voice-search/voice-search.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { render, unmountComponentAtNode } from 'preact-compat';
+import cx from 'classnames';
+import {
+  getContainerNode,
+  createDocumentationMessageGenerator,
+} from '../../lib/utils';
+import { component } from '../../lib/suit';
+import connectVoiceSearch from '../../connectors/voice-search/connectVoiceSearch';
+import VoiceSearch from '../../components/VoiceSearch/VoiceSearch';
+import defaultTemplates from './defaultTemplates';
+import { Template } from '../../types';
+
+const withUsage = createDocumentationMessageGenerator({ name: 'voice-search' });
+const suit = component('VoiceSearch');
+
+export type VoiceSearchCSSClasses = {
+  root: string;
+  button: string;
+  status: string;
+};
+
+export type VoiceSearchTemplates = {
+  buttonText: Template;
+  status: Template;
+};
+
+export type VoiceListeningState = {
+  status: string;
+  transcript?: string;
+  isSpeechFinal?: boolean;
+  errorCode?: string;
+};
+
+type VoiceSearchWidgetParams = {
+  container: string | HTMLElement;
+  cssClasses: VoiceSearchCSSClasses;
+  templates: VoiceSearchTemplates;
+  searchAsYouSpeak?: boolean;
+};
+
+type VoiceSearchWidgetRenderParams = {
+  isSupportedBrowser: () => boolean;
+  isListening: () => boolean;
+  toggleListening: (searchAsYouSpeak: boolean) => void;
+  voiceListeningState: VoiceListeningState;
+};
+
+const renderer = ({
+  container,
+  cssClasses,
+  templates,
+  searchAsYouSpeak,
+}: VoiceSearchWidgetParams) => ({
+  isSupportedBrowser,
```

Comment:
We can remove one level of indirection by getting these params from `widgetParams`.

Reply:
Nice! I had a hard time with that renderer and now it's clearer.